model/src/test/test_infectionsrtfeedback.py

```diff
@@ -0,0 +1,154 @@
+    I0_vec = np.array(I0_vec)
+    inf_feedback_strength = du.pad_x_to_match_y(
+        inf_feedback_strength, Rt, fill_value=inf_feedback_strength[0]
+    )
+    T = len(Rt)
+
+    Rt_adj = np.zeros(T)
+
+    for t in range(T):
+        Rt_adj[t] = Rt[t] * np.exp(
+            -inf_feedback_strength[t]
+            * np.dot(I0_vec, np.flip(inf_feedback_pmf))
+        )
+
+        I0_vec[t] = Rt_adj[t] * np.dot(I0_vec, np.flip(gen_int))
+
+    return {"infections": I0_vec, "rt": Rt_adj}
+
+
+def test_infectionsrtfeedback():
+    """
+    Test the InfectionsWithFeedback matching the Infections class.
+    """
+
+    Rt = jnp.array([0.5, 0.6, 0.7, 0.8])
+    I0 = jnp.array([1.0])
+    gen_int = jnp.array([0.25, 0.25, 0.25, 0.25])
+
+    # By doing the infection feedback strength 0, Rt = Rt_adjusted
+    # So infection should be equal in both
+    inf_feed_strength = DeterministicVariable(jnp.array([0.0]))
+    inf_feedback_pmf = DeterministicPMF(gen_int)
+
+    # Test the InfectionsWithFeedback class
+    InfectionsWithFeedback = latent.InfectionsWithFeedback(
+        infection_feedback_strength=inf_feed_strength,
+        infection_feedback_pmf=inf_feedback_pmf,
+    )
+
+    infections = latent.Infections()
+
+    with npro.handlers.seed(rng_seed=0):
+        samp1 = InfectionsWithFeedback.sample(
+            gen_int=gen_int,
+            Rt=Rt,
+            I0=I0,
+        )
+
+        samp2 = infections.sample(
+            gen_int=gen_int,
+            Rt=Rt,
+            I0=I0,
+        )
+
+    assert_array_equal(samp1.infections, samp2.infections)
+    assert_array_equal(samp1.rt, Rt)
+
+    return None
+
+
+def test_infectionsrtfeedback_feedback():
+    """
+    Test the InfectionsWithFeedback with feedback
+    """
+
+    Rt = jnp.array([0.5, 0.6, 0.7, 0.8])
+    I0 = jnp.array([1.0])
+    gen_int = jnp.array([0.25, 0.25, 0.25, 0.25])
+
+    inf_feed_strength = DeterministicVariable(jnp.repeat(0.5, len(Rt)))
+    inf_feedback_pmf = DeterministicPMF(gen_int)
+
+    # Test the InfectionsWithFeedback class
+    InfectionsWithFeedback = latent.InfectionsWithFeedback(
+        infection_feedback_strength=inf_feed_strength,
+        infection_feedback_pmf=inf_feedback_pmf,
+    )
+
+    infections = latent.Infections()
+
+    with npro.handlers.seed(rng_seed=0):
+        samp1 = InfectionsWithFeedback.sample(
+            gen_int=gen_int,
+            Rt=Rt,
+            I0=I0,
+        )
+
+        samp2 = infections.sample(
+            gen_int=gen_int,
+            Rt=Rt,
+            I0=I0,
+        )
+
+    res = _infection_w_feedback_alt(
+        gen_int=gen_int,
+        Rt=Rt,
+        I0=I0,
+        inf_feedback_strength=inf_feed_strength.sample()[0],
+        inf_feedback_pmf=inf_feedback_pmf.sample()[0],
+    )
+
+    assert not jnp.array_equal(samp1.infections, samp2.infections)
+    assert_array_equal(samp1.infections, res['infections'])
+    assert_array_equal(samp1.rt, res['rt'])
```

Comment:
@dylanhmorris, I've spent about 1.5 hours trying to solve this problem. The two pairs of arrays should match (but they don't). `samp1` was generated using the new `RandomVariable` `InfectionsWithFeedback`, and `res` was generated using a less efficient version of that implemented at the top of the script. Can you take a look?

Reply:
Will do.

Reply:
I see it. It's a sign thing. The manual implementation is positive `infection_feedback_strength` ==> downward feedback; it adds a `-`. The implementation in `sample_infections_with_feedback()` positive `infection_feedback_strength` ==> _upward_ feedback. There's no `-`.

Reply:
Changed the test and edited the documentation for `sample_infections_with_feedback()` in 38e7028

Reply:
All tests in this file now pass as of 52bfdb6, but that commit (intentionally) introduces breaking changes elsewhere that need addressing.

Reply:
All tests in this file now pass as of 52bfdb6, but to achieve that, this commit introduces breaking changes elsewhere that need addressing.